site/app/controllers/forum/ForumController.php

```diff
@@ -389,6 +390,9 @@ public function publishPost(){
         } else if(!$this->core->getQueries()->existsPost($thread_id, $parent_id)) {
             $this->core->addErrorMessage("There was an error submitting your post. Parent post doesn't exist in given thread.");
             $result['next_page'] = $this->core->buildUrl(array('component' => 'forum', 'page' => 'view_thread'));
+        } else if($this->core->getQueries()->isThreadLocked($thread_id) and $this->core->getUser()->getGroup() > 1 ) {
```

Comment:
I would recommend using functions like `accessAdmin` to make it more readable.
Also, note that full access grader's group is 2.